pkg/storage/storage.go

```diff
@@ -80,69 +97,157 @@ func (p *storage) GetContainerMetrics(pods ...apitypes.NamespacedName) ([]api.Ti
 
 	timestamps := make([]api.TimeInfo, len(pods))
 	resMetrics := make([][]metrics.ContainerMetrics, len(pods))
+	for podIdx, pod := range pods {
+		contPoints, present := p.pods[pod]
+		if !present {
+			continue
+		}
 
-	for i, pod := range pods {
-		metricPoint, present := p.pods[pod]
+		_, present = p.prevPods[pod]
 		if !present {
 			continue
 		}
 
-		contMetrics := make([]metrics.ContainerMetrics, len(metricPoint.Containers))
-		var earliestTS *time.Time
-		for i, contPoint := range metricPoint.Containers {
-			contMetrics[i] = metrics.ContainerMetrics{
+		var (
+			contMetrics = make([]metrics.ContainerMetrics, len(contPoints))
+			earliestTS  time.Time
+			window      time.Duration
+			contIdx     int
+		)
+		for contName, contPoint := range contPoints {
+			prevContPoint, present := p.prevPods[pod][contName]
```

Comment:
Instead of accessing map twice `p.prevPods[pod][contName]`, let's store it outside the loop and just access map once `prevPod[contName]`